### PR TITLE
Wire IceStorm metrics in transient mode

### LINKED ISSUE #5960: [Audit-Low] IceStorm transient mode has no topic metrics and no observer updater
`TransientTopicImpl` has no observer and never calls `published()`/`forwarded()`; `setObserverUpdater` is wired only in the persistent manager ([`TopicManagerI.cpp:230-234`](https://github.com/zeroc-ice/ice/blob/a53cc349e0379aa4f3a4c67160e4452982b6fd80/cpp/src/IceStorm/TopicManagerI.cpp#L230-L234)), so in transient mode topic metrics are absent and subscriber observers are never refreshed on metrics-view changes.

**Fix:** wire the transient path (effectively a feature addition) or document the limitation.

---
Split out of #5858 (item D7) for individual milestone triage.

*Filed from an automated AI correctness audit of `cpp/src/IceStorm` (Claude Fable 5, 2026-07-07); confirmed through a multi-agent adversarial refute pass over the actual source.*

### PR DESCRIPTION
**Pull request by pepone**
> [!NOTE]
> Stacked on #5965 (the first two commits belong to that PR, since this change uses the new `forwarded(count)` notification). I'll rebase once #5965 merges.

Transient IceStorm reported no topic metrics and never refreshed observers:

- `TransientTopicImpl` never attached a topic observer, so transient topics reported no `published`/`forwarded` metrics.
- `TransientTopicManagerImpl` did not register an observer updater, so topic and subscriber observers were never refreshed when a metrics view changed — subscriber observers (attached at creation by the shared `Subscriber` code) went stale.

This mirrors the persistent implementation: the transient topic attaches its observer at construction, notifies it from `publish`, detaches it on destroy and shutdown, and implements `updateObserver`/`updateSubscriberObservers`; the transient topic manager implements `Instrumentation::ObserverUpdater` and is registered with `setObserverUpdater` at both construction sites in `Service.cpp` (standalone transient mode and the IceGrid-embedded instance). Registration lives in `Service.cpp` rather than a `create()` factory because the transient manager is itself the servant and already has a Slice `create` operation.

Tested with the transient variants of `IceStorm/single`, `IceStorm/federation`, and `IceStorm/createOrRetrieve` (all green).

Fixes #5960.

### CODE REVIEW

File: cpp/src/IceStorm/TransientTopicI.cpp [hunk 1]
```diff
@@ -467,4 +485,28 @@ TransientTopicImpl::shutdown()
...
+
+    _observer.detach();
+}
+
+void
+TransientTopicImpl::updateObserver()
+{
+    lock_guard lock(_mutex);
+
+    if (_instance->observer())
```
[the remark this answers is not in the document]
Comment (pepone) [reply cpp/src/IceStorm/TransientTopicI.cpp]: 🤖 **Claude:** Done in 988e8b8eaf — `updateObserver` now skips a topic that was destroyed but not reaped yet.

File: cpp/src/IceStorm/Service.cpp [hunk 2]
```diff
@@ -110,6 +110,10 @@ ServiceI::start([[maybe_unused]] const string& serviceName, const CommunicatorPt
         try
         {
             _transientManager = make_shared<TransientTopicManagerImpl>(_instance);
+            if (_instance->observer())
+            {
+                _instance->observer()->setObserverUpdater(_transientManager);
+            }
```
[the remark this answers is not in the document]
Comment (pepone) [reply cpp/src/IceStorm/Service.cpp:116]: 🤖 **Claude:** Declining for this PR. There is no IceStorm metrics test for the persistent path either, so a metrics harness (admin facet + view toggling + counter assertions) would be new test infrastructure covering both modes — out of proportion for a patch-release parity fix that mirrors long-standing persistent code. Worth considering as separate test work if IceStorm metrics coverage becomes a priority.

File: cpp/src/IceStorm/TransientTopicI.cpp [hunk 3]
```diff
@@ -467,4 +485,29 @@ TransientTopicImpl::shutdown()
...
+    _observer.detach();
+}
+
+void
+TransientTopicImpl::updateObserver()
+{
+    lock_guard lock(_mutex);
+
+    // Don't reattach the observer of a topic that was destroyed but not reaped yet.
+    if (!_destroyed && _instance->observer())
```
Comment (bernardnormier) [inline cpp/src/IceStorm/TransientTopicI.cpp:498]: This `!_destroyed` guard is the right call — note that the persistent `TopicImpl::updateObserver` doesn't have it, so a metrics-view update between destroy and reap re-attaches an observer to a destroyed persistent topic. Not this PR's problem, but worth mirroring the guard back into `TopicImpl::updateObserver` as a small follow-up.

**Review by bernardnormier — APPROVED** [review]
LGTM. Verified this is a faithful mirror of the persistent implementation (attach in the constructor, notify in `publish` under the same lock that snapshots the subscribers, detach in `destroy`/`shutdown` under `_mutex`, updater registered at both transient construction sites — which are mutually exclusive with the persistent registration in `TopicManagerImpl::create`), that `Instance` creates the `TopicManagerObserverI` in transient mode too, and that the manager→topic lock ordering matches the existing reap path.
